packages/flutter/lib/src/widgets/basic.dart

```diff
@@ -4497,66 +4497,452 @@ class PositionedDirectional extends StatelessWidget {
+///
+///    If [crossAxisAlignment] is set to anything other than
+///    [CrossAxisAlignment.stretch], apply any constraints in the
+///    _cross axis_ direction that may have been passed down by the
+///    parent of the [Flex] without forcing the child to be any larger than
+///    it's stated size in that direction. This is called using _"loose
+///    constraints"_ (in this case, in the cross axis).
+///    Note that these are constraints coming from the _parent_ of the
+///    [Flex], not the [CrossAxisAlignment] of the [Flex].
+///
+///    However, if the [CrossAxisAlignment] of the [Flex] has been
+///    set to [CrossAxisAlignment.stretch] then force all the children of the
+///    [Flex] to be as large as possible in the cross axis direction. This is
+///    achieved by using _"tight constraints"_. In such a case, the maximum
+///    cross axis size of the children of the [Flex] will be forced to be the
+///    same as the maximum constraint (size) _in that direction_ (height or
+///    width) that was passed down by the parent of the [Flex].
+///
+///    If the parent of the [Flex] has not set any size limit in the cross
+///    axis, and the child is trying to be the max possible size in that
+///    axis, then the error you will see is likely _"BoxConstraints forces an
+///    infinite (height or width)"_.
+///
+/// 2. Children that are not [Flexible]s or [Expanded]s have a fixed size
+///    and so do not have a flex parameter. They will not go through steps
+///    2 or 3.
+///
+///    In the main axis, take the space left over and divide it among the
+///    [Flexible]s. (Again, [Expanded] is just a kind of [Flexible]).
+///
+///    The [Flexible.flex] parameter determines how much of the
+///    leftover space each [Flexible] will have reserved
+///    for it. Notice _"reserved for it"_ does not mean how big
+///    it's going to be. It means the space (or size) reserved for it
+///    is the maximum size the [Flexible] _could be_, in the main axis.
+///
+///    Add up all the [flex] values of each [Flexible]'s fit parameter,
+///    and each [Flexible] will have a percentage of the leftover space
+///    reserved for it that will be equal to the percentage
+///    of total [flex] each [Flexible] has. For example:
+///
+///    [Flexible] 1 has [flex] of 50.
+///    [Expanded] 1 has [flex] of 30.
+///    [Expanded] 2 has [flex] of 20.
+///
+///    In the above case, the total of all flex parameters is 100. The
+///    [Flexible] will get 50/100 (50%) of the remaining space reserved
+///    for it. The first [Expanded] will get 30/100 (30%) reserved for it,
+///    and the second one will get 20% of the remaining space
+///    reserved for it.
+///
+///    The numbers you use for your [Flexible.fit] parameters don't have to
+///    add to 100, the process remains the same. Making them total 100
+///    is simply easier to visualize.
+///
+/// 3. _Keep in mind that all of step 3 applies only to the main axis._
```

Comment:
```suggestion
/// 3. _Keep in mind that all of step three applies only to the main axis._
```

Nit: I tend to prefer using Strunk and White rules for small numbers: it's nicer to read.

Reply:
It's also proper grammar!